anti-tracking-threat-model.md

```diff
@@ -0,0 +1,196 @@
+1. Whether it will eventually be possible to frustrate the [attacker goals that
+   we want to frustrate](#attacker-goals-that-we-want-to-frustrate).
+1. Whether the mitigations actually mitigate the attacks.
+1. Whether the costs of the mitigations are worth the benefits.
+1. Probably other things.
+
+Nonetheless, we feel it's important to start with a concrete proposal in mind
+when discussing how to evolve both this threat model and the Web Packaging
+proposals.
+
+Often when we criticize new, technically distinct tracking vectors, we are told
+that “you can track users in so many ways so why care about this one?” In the
+case of signed packages we hear about means of tracking such as doctored links
+where cross-site tracking is built into the URL, or server-side exchanges of
+personally identifiable information such as users' email addresses.
+
+Browsers are working hard to prevent cross-site tracking, including with new
+limits and restrictions on old technologies. Web Packaging must not add to that work, although it's acceptable for Web Packaging to prevent tracking only after that other work has been done.
+
+Finally, the success of new web technologies, including signed packages, relies
+on better security and privacy guarantees than what we've had in the past. We
+want progression in this space, not the status quo.
+
+## The Actors
+
+* **The user.** This is the human who relies on the user agent to protect their
+  privacy.
+* **The user agent.** This is the web browser that tries to protect the user's
+  privacy.
+* **Distributor.** An entity that delivers a signed package to either the user or another distributor.
+  * **AdTech or `adtech.example`.** This is a distributor that the user also
+    engages with as a first-party site and that has a financial interest in
+    1. knowing what the user does on other websites to augment its rich profile
+       of the user and
+    2. individual targeting of ads, based on its rich profile of the user.
+* **Publisher.** An entity that owns a domain and publishes content there. The
+  publisher may not actually author the content, but they put their name on it.
+  * **News or `news.example`.** This is a publisher for a news website which
+    wants its articles to be served as signed packages with the user agent's URL
+    bar showing `news.example`.
+
+## Use Cases
+
+See https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html
+for use cases. It's possible some mitigations will sacrifice some use cases, but
+those mitigations should call out the use cases they break.
+
+## Attacker Capabilities
+
+1. AdTech has significant first-party traffic which means most users have an
+   `adtech.example` cookie holding a unique ID, even in browsers with
+   multi-keyed caches.
+1. AdTech can convince News to give them a `news.example` certificate's private
+   key or to tell a CA that AdTech has permission to receive certificates valid
+   to sign exchanges for `news.example`.
```

Comment:
We should add that AdTech can help or convince News to set up a signing *service* as an alternative to handing over keys. With such a service, AdTech can create the package, calculate its hash, and send the hash over for signing, all on the fly.

Reply:
I think there’s value in continuing to separate these concepts.

It would seem that your threat considerations consider the following scenarios:
1. AdTech creates a private key and obtains a certificate on News’ behalf (e.g. as CDNs commonly do)
1. News generates the key and then shares with AdTech
1. News holds the private key, but then signs on AdTech’s behalf (as commonly practiced by CDNs, typically called Split-TLS)
1. TLS Delegated Creds have a fourth model, where News generates the key, and then authorizes a key for AdTech

These each seem to represent different combinations, with the last being particularly relevant in the discussion of certificates and keys

Reply:
I've elaborated this section with @sleevi's 4 possibilities. I think they mostly differ in how visible AdTech's misbehavior would be to News, and how long that misbehavior could keep affecting users after News terminated the relationship with AdTech.

I agree that we need to be precise about which of "certificate" or "private key" is correct in each location, but because they mean different things, I don't think we can talk about just one of them. I'm using phrasing that implies that a certificate signs the package, which matches https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#cross-origin-cert-req, but strictly a private key (or keypair?) signs the package and the certificate just vouches for the authority of that key. I'm happy to take better wording.